File: simulador.java
public class simulador {
      private int monto;
      private int cuotas;
      private int mespago;
      private int diapago;
      private boolean seguro;

      private double interesmensual;
      private double factorinteres;
      private double total;
      private double valorcuota;

      public simulador(int monto,int cuotas, int mespago,int diapago, boolean seguro){
        this.factorinteres=0.05;
        this.monto=monto;
        this.cuotas=cuotas;
        this.mespago=mespago;
        this.diapago=diapago;
        this.seguro=seguro;
      }
      /////Getters
      public int getmonto(){
        return monto;
      }
      public int getcuotas(){
        return cuotas;
      }
      public int getmespago(){
        return mespago;
      }
      public int getdiapago(){
        return diapago;
      }
      public boolean getseguro(){
        return seguro;
      }

      public double getinteresmensual(){
        return interesmensual;
      }
      public double getfactorinteres(){
        return factorinteres;
      }
      public double gettotal(){
        return total;
      }
      public double getvalorcuota(){
        return valorcuota;
      }
      /////Setters
      public void setmonto(int monto){
        this.monto=monto;
      }
      public void setcuotas(int cuotas){
        this.cuotas=cuotas;
      }
      public void setmespago(int mespago){
        if (mespago>12||mespago<1){
            throw new IllegalArgumentException("del 1-12");
        }else{
        this.mespago=mespago;}
      }
      public void setdiapago(int diapago){
        if (this.mespago==1||this.mespago==3||this.mespago==5||this.mespago==7||this.mespago==8||this.mespago==10||this.mespago==12){
            if(diapago>31||diapago<1){
                throw new IllegalArgumentException("Este mes tiene 31 dias");
            }
        else if(this.mespago==2){
            if(diapago>29||diapago<1){
                throw new IllegalArgumentException("Este mes tiene 29 dias");
            }
        }else if(diapago>30||diapago<1){
            throw new IllegalArgumentException("Este mes tiene 30 dias");
            }
        }
        else{
        this.diapago=diapago;}
      }
      public void setseguro(boolean seguro){
        this.seguro=seguro;
      }
      /////
      public void simular(){       
        this.interesmensual=this.factorinteres*this.cuotas;
        this.valorcuota=(monto/cuotas)+(monto*interesmensual);
        this.total=(valorcuota*cuotas);
      }
      public void simular(int monto,int cuotas, int mespago,int diapago, boolean seguro){
        this.factorinteres=0.05;
        this.monto=monto;
        this.cuotas=cuotas;
        this.mespago=mespago;
        this.diapago=diapago;
        this.seguro=seguro;

        this.interesmensual=this.factorinteres*this.cuotas;
        this.valorcuota=(monto/cuotas)+(monto*interesmensual);
        this.total=(valorcuota*cuotas);

        }
    }

File: proyecto-credito-node-docker/src/yavascrip/simulador.js
class Simulador {
  constructor(monto, cuotas, mespago, diapago, seguro) {
    this.factorinteres = 0.05;
    this.monto = monto;
    this.cuotas = cuotas;
    this.mespago = mespago;
    this.diapago = diapago;
    this.seguro = seguro;

    this.interesmensual = 0;
    this.total = 0;
    this.valorcuota = 0;
  }

  // Getters
  getmonto() {
    return this.monto;
  }
  getcuotas() {
    return this.cuotas;
  }
  getmespago() {
    return this.mespago;
  }
  getdiapago() {
    return this.diapago;
  }
  getseguro() {
    return this.seguro;
  }
  getinteresmensual() {
    return this.interesmensual;
  }
  getfactorinteres() {
    return this.factorinteres;
  }
  gettotal() {
    return this.total;
  }
  getvalorcuota() {
    return this.valorcuota;
  }

  // Setters
  setmonto(monto) {
    this.monto = monto;
  }
  setcuotas(cuotas) {
    this.cuotas = cuotas;
  }
  setmespago(mespago) {
    if (mespago > 12 || mespago < 1) {
      throw new Error("del 1-12");
    } else {
      this.mespago = mespago;
    }
  }
  setdiapago(diapago) {
    const m = this.mespago;
    if ([1, 3, 5, 7, 8, 10, 12].includes(m)) {
      if (diapago > 31 || diapago < 1) {
        throw new Error("Este mes tiene 31 dias");
      } else {
        this.diapago = diapago;
      }
    } else if (m === 2) {
      if (diapago > 29 || diapago < 1) {
        throw new Error("Este mes tiene 29 dias");
      } else {
        this.diapago = diapago;
      }
    } else {
      if (diapago > 30 || diapago < 1) {
        throw new Error("Este mes tiene 30 dias");
      } else {
        this.diapago = diapago;
      }
    }
  }
  setseguro(seguro) {
    this.seguro = seguro;
  }

  simular() {
    this.interesmensual = this.factorinteres * this.cuotas;
    this.valorcuota = this.monto / this.cuotas + this.monto * this.interesmensual;
    this.total = this.valorcuota * this.cuotas;
  }

  simularConParametros(monto, cuotas, mespago, diapago, seguro) {
    this.factorinteres = 0.05;
    this.monto = monto;
    this.cuotas = cuotas;
    this.mespago = mespago;
    this.diapago = diapago;
    this.seguro = seguro;

    this.interesmensual = this.factorinteres * this.cuotas;
    this.valorcuota = this.monto / this.cuotas + this.monto * this.interesmensual;
    this.total = this.valorcuota * this.cuotas;
  }
}
module.exports = Simulador; 